File: AddDigits.java
package leetCode;

public class AddDigits {
	public int addDigits(int num) {
        int result = 0;
        while (num > 0){
        	int cur_digit = num % 10;
        	num /= 10;
        	result += cur_digit;
        }
        if (result < 10){
        	return result;
        } else {
        	return addDigits(result);
        }
    }
}

File: python_solutions/add_digits.py
class Solution(object):
    def addDigitsOrig(self, num):
        """
        :type num: int
        :rtype: int
        """
        if num < 10:
        	return num

        num = str(num)
        total = 0
        for d in num:
        	total += int(d)
        return self.addDigits(total)

    def addDigits(self, num):
    	if num < 10:
            return num
        
        total = 0
        while num > 0:
        	total += num % 10
        	num /= 10

        return self.addDigits(total)

s = Solution()
print s.addDigits(10)